lib/spack/spack/bootstrap.py

```diff
@@ -837,3 +838,156 @@ def ensure_flake8_in_path_or_raise():
+    }
+    if platform.system().lower() == 'linux':
+        _core_system_exes['xz'] = _missing_executable(
+            'xz', 'compress/decompress code archives'
+        )
+
+    # Executables that are not bootstrapped yet
+    result = [functools.partial(_system_executable, exe, msg)
+              for exe, msg in _core_system_exes.items()]
+    # Python modules
+    result += [
+        functools.partial(
+            _required_python_module, 'clingo', clingo_root_spec(),
+            'MISSING: "clingo" python module (required to concretize specs)'
+        )
+    ]
+    return result
+
+
+def _buildcache_requirements():
+    _buildcache_exes = {
+        'file': _missing_executable('file', 'analyze files for buildcaches'),
+        ('gpg2', 'gpg'): _missing_executable('gpg2', 'sign/verify buildcaches')
+    }
+    if platform.system().lower() == 'darwin':
+        _buildcache_exes['otool'] = _missing_executable('otool', 'relocate binaries')
+
+    # Executables that are not bootstrapped yet
+    result = [functools.partial(_system_executable, exe, msg)
+              for exe, msg in _buildcache_exes.items()]
+
+    if platform.system().lower() == 'linux':
+        result.append(functools.partial(
+            _required_executable, 'patchelf', patchelf_root_spec(),
+            _missing_executable('patchelf', 'relocate binaries')
+        ))
+
+    return result
+
+
+def _optional_requirements():
+    _optional_exes = {
+        'zstd': _missing_executable('zstd', 'compress/decompress code archives'),
+        'svn': _missing_executable('svn', 'manage subversion repositories'),
+        'hg': _missing_executable('hg', 'manage mercurial repositories')
+    }
+    # Executables that are not bootstrapped yet
+    result = [functools.partial(_system_executable, exe, msg)
+              for exe, msg in _optional_exes.items()]
+    return result
+
+
+def _development_requirements():
+    return [
+        functools.partial(
+            _required_executable, 'isort', isort_root_spec(),
+            'MISSING: "isort" executable (required for style checks)'
+        ),
+        functools.partial(
+            _required_executable, 'mypy', mypy_root_spec(),
+            'MISSING: "mypy" executable (required for style checks)'
+        ),
+        functools.partial(
+            _required_executable, 'flake8', flake8_root_spec(),
+            'MISSING: "flake8" executable (required for style checks)'
+        ),
+        functools.partial(
+            _required_executable, 'black', black_root_spec(),
+            'MISSING: "black" executable (required for code formatting)'
+        )
+    ]
+
+
+def status_message(section):
+    """Return a status message to be printed to screen that refers to the
+    section passed as argument.
+
+    Args:
+        section (str): either 'core' or 'buildcache' or 'optional' or 'develop'
+    """
+    ok_token, ko_token = "\U0001F44D", "\U0001F44E"
```

Comment:
I like the output on a mac terminal but note the tokens don't come across well in some shells (e.g., `xterm` on a mac):

```
Spack v0.17.0 - python@3.7 
  Core Functionalities
ð Binary packages
ð
ð Optional Features
  MISSING: "zstd" executable (required to compress/decompress code archives)
  Development Dependencies
ð
```

Reply:
Yeah, I think another PR proposed unicode emoji and we rejected it for this reason.

Reply:
Ugh, I didn't try xterm. I'll revert to colorized `[PASS]` and `[FAIL]` for the time being :slightly_frowning_face:

Reply:
Done in b1fddcd